libcentrifugo/engine/engineredis/engine.go

```diff
@@ -356,38 +361,133 @@ func getConfig(getter config.Getter) *Config {
 		}
 	}
 
+	password := getter.GetString("redis_password")
+	db := getter.GetString("redis_db")
+
 	if len(sentinelAddrs) > 0 && masterName == "" {
 		logger.FATAL.Fatalln("Redis master name required when Sentinel used")
 	}
 
-	conf := &Config{
-		Host:           getter.GetString("redis_host"),
-		Port:           getter.GetString("redis_port"),
-		Password:       getter.GetString("redis_password"),
-		DB:             getter.GetString("redis_db"),
-		URL:            getter.GetString("redis_url"),
-		PoolSize:       getter.GetInt("redis_pool"),
-		API:            getter.GetBool("redis_api"),
-		NumAPIShards:   getter.GetInt("redis_api_num_shards"),
-		Prefix:         getter.GetString("redis_prefix"),
-		MasterName:     masterName,
-		SentinelAddrs:  sentinelAddrs,
-		ConnectTimeout: time.Duration(getter.GetInt("redis_connect_timeout")) * time.Second,
-		ReadTimeout:    time.Duration(getter.GetInt("redis_read_timeout")) * time.Second,
-		WriteTimeout:   time.Duration(getter.GetInt("redis_write_timeout")) * time.Second,
+	var shardConfigs []*Config
+
+	if len(hosts) < numShards {
+		newHosts := make([]string, numShards)
+		for i := 0; i < numShards; i++ {
+			newHosts[i] = hosts[0]
```

Comment:
This is mostly for situation when configuration look like this:

```
centrifugo --engine=redis --redis_host=127.0.0.1 --redis_port=6379,6380,6381
```

So 3 shards look like this: 127.0.0.1:6379,  127.0.0.1:6380,  127.0.0.1:6381.

But actually this piece relates to my specific concerns you asked. When I wrote it I see how simple it would be to add just one more host or port. But first of all it look ugly internally.

And what if someone will write

```
centrifugo --engine=redis --redis_host=127.0.0.1,127.0.0.2 --redis_port=6379,6380,6381
```

I don't think we need to try to guess what user meant, right? So maybe adding extra check here? Like this:
```go
if len(hosts) < numShards && len(hosts) == 1 {
    newHosts := make([]string, numShards)
    for i := 0; i < numShards; i++ {
        newHosts[i] = hosts[0]
    }
    hosts = newHosts
} else {
    FATAL('Wrong sharding configuration!')
}
```

Will write more thoughts about sharding in comment under pull request.